src/charm.py

```diff
@@ -80,10 +82,52 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _create_service_account(self):
+        """Create a ServiceAccount for training-operator."""
+        self.unit.status = MaintenanceStatus("Creating ServiceAccount")
+        try:
+            K8sResources.create_service_account()
+        except ResourceHandlingFailed as e:
+            logging.error(
+                "Unable to create ServiceAccount for training-operator: %s", str(e)
+            )
```

Comment:
consider returning something from this function, like a `bool` or similar to indicate success or failure? Right now this essentially silently fails unless you're watching the logs. Same for other helpers.